src/awkward/nplike.py

```diff
@@ -233,6 +243,10 @@ def broadcast_to(self, *args, **kwargs):
         # array, shape
         return self._module.broadcast_to(*args, **kwargs)
 
+    def append(self, *args, **kwargs):
+        # array, element
+        return self._module.append(*args, **kwargs)
```

Comment:
I think CuPy does have an implementation of append [here](https://docs.cupy.dev/en/latest/reference/generated/cupy.append.html).

Reply:
I don't see it, and I just upgraded from CuPy 8.5.0 to 8.6.0 (the latest, apparently):

```python
>>> import cupy
>>> cupy.__version__
'8.6.0'
>>> cupy.append
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'cupy' has no attribute 'append'
```

Reply:
Hey, I completely forgot that we weren't putting a `CuPy` minimum version in `requirements-dev.txt` because Azure didn't have GPU's and the pip install would fail anyways. So this PR should be mergeable once the tests pass!